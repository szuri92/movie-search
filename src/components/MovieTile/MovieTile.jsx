import React from 'react'
import { useHistory } from 'react-router-dom'

import styles from './MovieTile.module.scss'

import Image from '../Image'

const MovieTile = ({ movie } = {}) => {
    const history = useHistory()

    const seeMovieDetails = () => {
        history.push(`/detail/${movie.id}`)
    }

    return (
        <div className={styles.movie_tile_box} onClick={seeMovieDetails}>
            <Image src={movie.poster_path}/>
            <h1 className={styles.title}>{ movie.title }</h1>
            <div className={styles.overview}>
                { movie.overview }
            </div>
        </div>
    )
}

export default MovieTile