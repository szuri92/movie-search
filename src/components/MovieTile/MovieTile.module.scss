.movie_tile_box {
    padding: 10px;
    border: 1px solid #dedede;
    width: 342px;
    padding: 15px;

    .title {
        text-align: left
    }

    .overview {
        overflow: hidden;
        text-overflow: ellipsis;
        display: -webkit-box;
        -webkit-line-clamp: 4; 
        -webkit-box-orient: vertical;
        text-align: left;
    }
}