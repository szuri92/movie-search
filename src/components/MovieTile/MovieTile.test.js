import React from "react"
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import MovieTile from './MovieTile'

const movieMock = {
    id: 1,
    overview: 'overview',
    poster_path: 'poster_path',
}

describe('MovieTile', () => {
  it('renders correctly', () => {
    const movieTitle = shallow(<MovieTile movie={movieMock}/>)
    expect(shallowToJson(movieTitle)).toMatchSnapshot()
  })
})