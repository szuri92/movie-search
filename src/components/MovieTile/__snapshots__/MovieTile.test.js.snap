// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile renders correctly 1`] = `
<div
  className="movie_tile_box"
  onClick={[Function]}
>
  <Image
    src="poster_path"
  />
  <h1
    className="title"
  />
  <div
    className="overview"
  >
    overview
  </div>
</div>
`;
